src/safeds/data/labeled/containers/_time_series_dataset.py

```diff
@@ -0,0 +1,323 @@
+    A time series dataset maps feature and time columns to a target column. Not like the TableDataset a TimeSeries needs contain one target and one time column, but can have empty features.
+
+    Create a tabular dataset from a mapping of column names to their values.
+
+    Parameters
+    ----------
+    data:
+        The data.
+    target_name:
+        Name of the target column.
+    time_name:
+        Name of the time column.
+    extra_names:
+        Names of the columns that are neither features nor target. If None, no extra columns are used, i.e. all but
+        the target column are used as features.
+
+    Raises
+    ------
+    ColumnLengthMismatchError
+        If columns have different lengths.
+    ValueError
+        If the target column is also an extra column.
+    ValueError
+        If no feature columns remains.
+
+    Examples
+    --------
+    >>> from safeds.data.labeled.containers import TabularDataset
+    >>> dataset = TimeSeriesDataset(
+    ...     {"id": [1, 2, 3], "feature": [4, 5, 6], "target": [1, 2, 3], "error":[0,0,1]},
+    ...     target_name="target",
+    ...     time_name = "id",
+    ...     extra_names=["error"]
+    ... )
+    """
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Dunder methods
+    # ------------------------------------------------------------------------------------------------------------------
+    def __init__(
+        self,
+        data: Table | Mapping[str, Sequence[Any]],
+        target_name: str,
+        time_name: str,
+        extra_names: list[str] | None = None,
+    ):
+        # Preprocess inputs
+        if not isinstance(data, Table):
+            data = Table(data)
+        if extra_names is None:
+            extra_names = []
+
+        # Derive feature names
+        feature_names = [name for name in data.column_names if name not in {target_name, *extra_names, time_name}]
+
+        # Validate inputs
+        if time_name in extra_names:
+            raise ValueError(f"Column '{time_name}' cannot be both time and extra.")
+        if target_name in extra_names:
+            raise ValueError(f"Column '{target_name}' cannot be both target and extra.")
+        if len(feature_names) == 0:
+            feature_names = []
+
+        # Set attributes
+        self._table: Table = data
+        self._features: Table = data.keep_only_columns(feature_names)
+        self._target: Column = data.get_column(target_name)
+        self._time: Column = data.get_column(time_name)
+        self._extras: Table = data.keep_only_columns(extra_names)
+
+    def __eq__(self, other: object) -> bool:
+        """
+        Compare two tabular datasets.
+
+        Returns
+        -------
+        equals:
+            'True' if features and targets are equal, 'False' otherwise.
+        """
+        if not isinstance(other, TimeSeriesDataset):
+            return NotImplemented
+        return (
```

Comment:
Small performance improvement:
```suggestion
        return (self is other) or (
```

Reply:
added